### PR TITLE
Misc ACI deployments changes

### PR DESCRIPTION
**Pull request by jumaffre**
This makes the ACI deployment scripts more modular so that we can easily deploy different types of containers in the future (e.g. node runtime container, attestation container). We now also use a virtual environment to install the Azure Python SDK libraries so that we always use a recent, known version.

### CODE REVIEW

File: scripts/azure_deployment/arm_aci.py [hunk 1]
```diff
@@ -98,87 +163,80 @@ def make_aci_deployment(parser: ArgumentParser) -> Deployment:
...
+        "type": "Microsoft.ContainerInstance/containerGroups",
+        "apiVersion": "2022-04-01-preview",
+        "name": args.deployment_name,
+        "location": args.region,
+        "properties": container_group_properties,
+    }
+
+    arm_template["resources"].append(container_group)
+
+    # print(json.dumps(arm_template, indent=2))
```
Comment (DomAyre) [inline scripts/azure_deployment/arm_aci.py:235]: Can be cleaned up

**Review by DomAyre — APPROVED** [review]
Aside from commented line looks good to me

**Review by takuro-sato — COMMENTED** [review]
I'll still checking but can you edit `attestation_container_canary` to run the attestation container CI? I think it's better to add `scripts/azure_deployment` to .azure-pipelines-attestation-container.yml and to .azure_pipelines_snp.yml (It's out of this PR's scope though)

File: scripts/azure_deployment/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+azure-identity==1.12.0
```
[the remark this answers is not in the document]
Comment (DomAyre) [reply scripts/azure_deployment/requirements.txt:1]: That would mean we'd get updates under our feet so I think this approach is more stable

**Pull request merged**